resource/docker/container/preparer.go

```diff
@@ -120,6 +120,12 @@ func (p *Preparer) Prepare(render resource.Renderer) (resource.Task, error) {
 }
 
 func validateContainer(container *Container) error {
+	if container.Name == "" {
```

Comment:
see #337. `required` is not enough to prevent someone from explicitly setting a field to an empty string